frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/BrowseNavSection.tsx

```diff
@@ -29,59 +29,71 @@ export const BrowseNavSection = ({
     "expand-browse-in-nav",
   );
 
+  const [opened, { toggle }] = useDisclosure(expandBrowse);
+
   const entityTypes = useSelector(
     (state) => getEmbedOptions(state).entity_types,
   );
   const isEmbeddingIframe = useSelector(getIsEmbeddingIframe);
 
+  const handleToggle = () => {
+    toggle();
+    setExpandBrowse(!opened);
+  };
+
   return (
-    <CollapseSection
-      header={
+    <div aria-selected={opened} role="tab">
+      <Group
+        align="center"
+        gap="0.5rem"
+        onClick={handleToggle}
+        component="button"
```

Comment:
Consider using [UnstyledButton](https://v7.mantine.dev/core/unstyled-button/) from the ui library